client/my-sites/comments/comment/comment-author.jsx

```diff
@@ -0,0 +1,122 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'gridicons';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Emojify from 'components/emojify';
+import ExternalLink from 'components/external-link';
+import Gravatar from 'components/gravatar';
+import { convertDateToUserLocation } from 'components/post-schedule/utils';
+import { gmtOffset, timezone } from 'lib/site/utils';
+import { urlToDomainAndPath } from 'lib/url';
+import { getSiteComment } from 'state/selectors';
+import { getSelectedSite, getSelectedSiteId } from 'state/ui/selectors';
+import { getAuthorDisplayName, getPostTitle } from 'my-sites/comments/comment/utils';
+
+export class CommentAuthor extends Component {
+	static propTypes = {
+		commentId: PropTypes.number,
+		isExpanded: PropTypes.bool,
+	};
+
+	render() {
+		const {
+			authorDisplayName,
+			authorUrl,
+			commentDate,
+			commentType,
+			commentUrl,
+			gravatarUser,
+			isExpanded,
+			moment,
+			postTitle,
+			site,
+		} = this.props;
+
+		const localizedDate = convertDateToUserLocation(
+			commentDate || moment(),
+			timezone( site ),
+			gmtOffset( site )
+		);
+
+		const formattedDate = localizedDate.format( 'll LT' );
+
+		const relativeDate = moment()
+			.subtract( 1, 'month' )
+			.isBefore( localizedDate )
+			? localizedDate.fromNow()
+			: localizedDate.format( 'll' );
+
+		return (
+			<div className="comment__author">
+				<div className="comment__author-avatar">
+					{ 'comment' === commentType && <Gravatar user={ gravatarUser } /> }
+					{ 'comment' !== commentType && <Gridicon icon="link" size={ 24 } /> }
```

Comment:
Nitpick: a comment here explaining the non-`comment` `comentType`s could be helpful. :)